waltid-services/waltid-wallet-api/src/main/kotlin/id/walt/webwallet/web/controllers/exchange/ExchangeExternalSignaturesController.kt

```diff
@@ -79,111 +84,148 @@ fun Application.exchangeExternalSignatures() = walletRoute {
                             "vpTokenParams object that is contained within."
                     body<PrepareOID4VPResponse> {
                         required = true
+                        example(
+                            "W3C Verifiable Credential",
+                            ExchangeExternalSignaturesExamples.prepareOid4vpResponseW3CVCExample(),
+                        )
+                        example(
+                            "W3C SD-JWT Verifiable Credential",
+                            ExchangeExternalSignaturesExamples.prepareOid4vpResponseW3CSDJWTVCExample(),
+                        )
+                        example(
+                            "IETF SD-JWT Verifiable Credential",
+                            ExchangeExternalSignaturesExamples.prepareOid4vpResponseIETFSDJWTVCExample(),
+                        )
                     }
                 }
             }
         }) {
             val walletService = getWalletService()
+            runCatching {
```

Comment:
I don't think `runCatching` is required here, since any exception would be caught by [StatusPages](https://github.com/walt-id/waltid-identity/blob/de6ebe904f0c6516b35fdc2988e1f7d33f3e90f7/waltid-services/waltid-service-commons/src/main/kotlin/id/walt/commons/web/plugins/StatusPages.kt#L26) from service-commons and handled accordingly

Reply:
I think it's much cleaner in the context of the handler to have it like this, as the response can be computed according to what is thrown. Also, in general, I am against introducing assumptions about what other components do, or handle, when not necessary. StatusPages are a relatively "new thing" and we'll see how they will turn out...

Reply:
agree with the assumptions aspect
but here, the actual error is hidden behind the "Unknown" - bad-request. The recipient won't receive AuthorizationError, "ephemeral key", etc. But maybe this is also smth. that could be handled within the presentation error logic ticket.